### DISCUSSION TITLE
How to combine `git_files` and `oldfiles`?

### DISCUSSION THREAD

**Discussion by dycw** [ibhagwan/fzf-lua#1140] [Q&A] [answered]
Is it possible to do so? This intersection of the two file pickers would be incredibly convenient to have.

**Comment by ibhagwan** [comment, marked as answer]
It’s possible but you’d have to write your own code for it via the API, you’d need to write a contents function that runs `git ls-files` inputs all entries and then goes over the old files and does the same.

Since one is a shell command and the other is neovim command you’d need to use `vim.system`, performance might not be ideal as the external process currently used.

**Comment by dycw** [comment]
Thanks! After some thought, I've written something a bit different — what I wanted is `git ls-files` in the order they were used, i.e. MRU (most recently used)

https://github.com/dycw/dotfiles/commit/339f65779bde7c684477ffee87b40a289b040f8e

Gets the job done, but, care to share some thoughts on how I did it?


![Screenshot 2024-04-18 at 11 13 10 PM](https://github.com/ibhagwan/fzf-lua/assets/10572122/7eb5c478-74e1-4b61-ba96-8a39f31542cd)

**Comment by ibhagwan** [comment]
@dycw, looks good and clean, a few comments I would add:

- You can most likely replace both the git root / path modification with utilities from `fzf-lua.path` module.
- I generally prefer the asynchrony approach where I add files as they are found, that way the user doesn’t have to way for the git command to return from the shell, if you run your code on a very large repo neovim will freeze until `git ls-files` returns
- Given the above I also delegate the shell tasks to an external process via `fzf-lua.libuv` module, while that may be an overkill for your command you can probably still use `vim.fn.jobstart` for an asynchrony operation

Ovrrall, if you’re happy with the result and aren’t dealing with large repos this is perfect.
